test/client.js

```diff
@@ -0,0 +1,171 @@
+            });
+    });
+
+    it('should get cached access token with specified scope', () => {
+        let accounts_nock = nocks.token('client_credentials')
+            .reply(200, {access_token: 'jrrtoken'});
+
+        return accounts.generate_token(scopes)
+            .then(result => {
+                result.should.have.property('access_token');
+                result.access_token.should.be.a('string');
+                result.access_token.should.equal('jrrtoken');
+                accounts_nock.isDone().should.not.equal(true);
+
+                nock.cleanAll();
+            });
+    });
+
+    it('should not get cached access token when cached is cleared', () => {
+        let accounts_nock = nocks.token('client_credentials')
+            .reply(200, {access_token: 'jrrtoken'});
+
+        accounts.clear_cache('client');
+
+        return accounts.generate_token(scopes)
+            .then(result => {
+                result.should.have.property('access_token');
+                result.access_token.should.be.a('string');
+                result.access_token.should.equal('jrrtoken');
+                accounts_nock.isDone().should.equal(true);
+
+                nock.cleanAll();
+            });
+    });
+});
+
+describe('refresh_token', () => {
+
+    accounts.configure(configuration);
+
+    it('should properly refresh token', () => {
+        let accounts_nock = nocks.token('client_credentials')
+            .reply(200, {access_token: 'jrrtoken', refresh_token: 'jrrrefreshtoken'});
+
+        let accounts_refresh_nock = nocks.token('refresh_token')
+            .reply(200, {access_token: 'freshjrrtoken', refresh_token: 'jrrrefreshtoken'});
+
+        accounts.clear_cache('client');
+
+        return accounts.generate_token(scopes)
+            .then(result => {
+                result.should.have.property('access_token');
+                result.access_token.should.be.a('string');
+                result.access_token.should.equal('jrrtoken');
+                result.should.have.property('refresh_token');
+                result.refresh_token.should.be.a('string');
+                result.refresh_token.should.equal('jrrrefreshtoken');
+                accounts_nock.isDone().should.equal(true);
+
+                return result.refresh_token;
+            })
+            .then(accounts.refresh_token)
+            .then(result => {
+                result.should.have.property('access_token');
+                result.access_token.should.be.a('string');
+                result.access_token.should.equal('freshjrrtoken');
+                result.should.have.property('refresh_token');
+                result.refresh_token.should.be.a('string');
+                result.refresh_token.should.equal('jrrrefreshtoken');
+                accounts_refresh_nock.isDone().should.equal(true);
+
+                nock.cleanAll();
+            });
+    });
+
+    it('should properly update cache of refreshed token', () => {
```

Comment:
I think it would be better to also have a test for cache.js

Reply:
Done:
00f031a